### PR TITLE
Integration for https://github.com/vector-im/vector-web

### PR DESCRIPTION
**Pull request by mrdrogdrog**
This plugins adds integration for Vector Web - Chat for Matrix.org with custom URL

### CODE REVIEW

File: vectorweb/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module.exports = (Franz) => {
+  class VectorChat extends Franz {
+    validateServer(URL) {
+      const api = `${URL}`;
+      return new Promise((resolve, reject) => {
+        $.get(api, (resp) => {
+            resolve();
```
Comment (adlk) [inline vectorweb/index.js]: This validation will allow every working url like https://google.com which is not a working vector web interface.
Instead, I suggest you make use of the [config.json](https://vector.im/beta/config.json) that is provided by the vector web interface and just check if it exists, you can even look for specific parameters ([as we are doing in the docs](https://github.com/meetfranz/plugins/blob/develop/docs/integration.md#indexjs))

I haven't tested the code but you should get the idea.

```js
module.exports = (Franz) => {
  class VectorChat extends Franz {
    validateServer(URL) {
      const api = `${URL}/config.json`;
      return new Promise((resolve, reject) => {
        $.get(api, (resp) => {
          if (resp.status < 200 || resp.status >= 300) {
            reject();
          } else {
            resolve();
          }
        }).fail(reject);
      });
    }
  }

  return VectorChat;
};
```

**Comment by mrdrogdrog** [comment]
wait wait wait

**Comment by mrdrogdrog** [comment]
sry I forgot the response code check. Its now online.

**Comment by adlk** [comment]
I realized that for this simple kind of check it's not even necessary, because it will fail if the file does not exist anyway.

**Comment by mrdrogdrog** [comment]
okay.

**Pull request merged**